docs/virtual_machine_template_guide.adoc

```diff
@@ -0,0 +1,189 @@
+= VirtualMachine Templates
+
+VirtualMachine templates are openshift templates (https://docs.openshift.org/latest/dev_guide/templates.html) +
+that describe VirtualMachine specifications which produce the following template objects: +
+**One** OfflineVirtualMachine and PersistentVolumeClaim. +
+The following document describes the process of creating a Virtual machine template which will be managed by ManageIQ.
+
+== Uploading a Template
+
+If you have a YAML file that defines a VirtualMachine template, you can upload the template to an Openshift cluster using the CLI. 
+To upload a template to your current project’s template library, pass the YAML file with the following command: +
+
+----
+$ oc create -f <filename>
+----
+
+== Modifying an Uploaded Template
+You can edit a template that has already been uploaded to your project by using the following command: +
+
+----
+$ oc edit template <template>
+----
+
+Writing Templates
+You can define new VirtualMachine templates to make it easy to recreate all relevant objects (offlineVirtualMachine and PersistentVolumeClaim). The template will define the objects it creates along with some metadata to guide the creation of those objects.
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha1
+kind: Template
+metadata:
+  name: RHEL-7 ❶
+objects:
+- apiVersion: kubevirt.io/v1alpha1
+  kind: offlineVirtualMachine ❷
+  metadata:
+    name: rhel-7-${NAME} ❸
+  spec:
+    template:
+      metadata:
+      labels:
+        my: label
+      spec:
+        domain:
+          resources: ❹
+            memory: ${MEMORY} 
+            cpu: ${CPU}
+          devices:
+      disks: ➎
+        name: disk0
+        volumeName: mypvc 
+      volumes: ➏
+         name: mypvc
+         presistentVolumeClaim:
+      claimName: rhel-7-pvc-${NAME}
+- apiVersion: kubevirt.io/v1alpha1
+  kind: presistentVolumeClaim ❼
+  metadata:
+    name: rhel-7-pvc-${NAME} ❽
+labels: ...       
+parameters: ...
+
+----
+ 
+❶ The unique name of the template. +
```

Comment:
I normally tend to not use such special characters in markdown, but the result looks nice.

Reply:
It resembles openshift guides, if it will get lost in translation in the future then ill change it.